### PR TITLE
Initial working tf code

### PR DESCRIPTION
**Pull request by vaijab**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  additional_zones = [
+    "${data.google_compute_zones.available.names[1]}",
+    "${data.google_compute_zones.available.names[2]}",
+  ]
+
+  initial_node_count = "${var.initial_node_count}"
+
+  min_master_version = "1.8.6-gke.0"
+  enable_legacy_abac = "false"
```
Comment (asmckinlay) [inline main.tf]: what's this? what's abac?

Comment (asmckinlay) [reply main.tf]: would it not be false by default?

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  subnetwork = "nodes"
+
+  ip_allocation_policy {
+    cluster_secondary_range_name = "pods"
+    services_secondary_range_name = "services"
+  }
+
+  addons_config {
+    horizontal_pod_autoscaling {
+      disabled = true
```
Comment (asmckinlay) [inline main.tf]: why don't we want horizontal auto scaling? Unless we are manually defining the number to scale to??

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    kubernetes_dashboard {
+      disabled = true
```
Comment (asmckinlay) [inline main.tf]: dashboards are pretty though 😛

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,54 @@
+data "google_compute_zones" "available" {}
+
+resource "google_container_cluster" "cluster" {
+  name = "${var.name}"
+  zone = "${data.google_compute_zones.available.names[0]}"
```
Comment (asmckinlay) [inline main.tf]: could we have in a comment what the zone value of each `data.google_compute_zones.available.names[X]` is please?

File: main.tf (hunk 3 above)
Comment (asmckinlay) [reply main.tf]: Can ignore this, not important

File: main.tf (hunk 4 above)
Comment (vaijab) [reply main.tf]: It depends on the `region` that is set when configuring google provider outside of this module.

File: main.tf (hunk 1 above)
Comment (vaijab) [reply main.tf]: I believe it is enabled by default. ABAC is legacy authorization mode in kubernetes. It has been replaced by RBAC - https://kubernetes.io/docs/admin/authorization/

File: main.tf (hunk 2 above)
Comment (vaijab) [reply main.tf]: It's just easier to begin with a static cluster before we know what's going on. For autoscaler to work well, one would have to set resource limits properly, etc. We'll get there eventually.

File: main.tf (hunk 3 above)
Comment (vaijab) [reply main.tf]: :-D

**Comment by vaijab** [comment]
> It would be interesting, once you are done, to have a presentation of your rationale in all the 'implicit' choice that you have made

Sure thing.

File: main.tf (hunk 4 above)
Comment (asmckinlay) [reply main.tf]: ok :)

**Pull request merged**